src/main/kotlin/com/viartemev/thewhiterabbit/consumer/ConfirmConsumer.kt

```diff
@@ -36,12 +33,22 @@ class ConfirmConsumer internal constructor(private val AMQPChannel: Channel, AMQ
         )
     }
 
-    suspend fun consumeWithConfirm(handler: suspend (Delivery) -> Unit, handlerDispatcher: CoroutineDispatcher = Dispatchers.Default) = coroutineScope {
+    suspend fun consumeWithConfirm(handler: suspend (Delivery) -> Unit, handlerDispatcher: CoroutineDispatcher = Dispatchers.Default) {
         val delivery = continuations.receive()
+        val deliveryTag = delivery.envelope.deliveryTag
         withContext(handlerDispatcher) { handler(delivery) }
-        AMQPChannel.basicAck(delivery.envelope.deliveryTag, false)
+        try {
+            AMQPChannel.basicAck(deliveryTag, false)
+        } catch (e: Exception) {
+            val errorMessage = "Can't ack a message with deliveryTag: $deliveryTag"
+            logger.error { errorMessage }
+            throw RuntimeException(errorMessage)
```

Comment:
I think we should throw some concrete exception here for user to be able to recover from error somehow.